gp-includes/routes/translation.php

```diff
@@ -278,6 +279,12 @@ public function translations_post( $project_path, $locale_slug, $translation_set
 
 			if ( $this->can( 'approve', 'translation-set', $translation_set->id ) || $this->can( 'write', 'project', $project->id ) ) {
 				$set_status = 'current';
+				$user_id    = gp_post( 'user_id' );
```

Comment:
This allows assigning any `user_id` to a translation. Wouldn't it be a more viable approach to add a flag that simply doesn't override the `user_id` further up in the code?

Reply:
I started implementing this using a `keep_user` flag but it was tricky to retrieve the user_id for the current translation at this point in the code. If there is a function to do this I'd be happy to use it but I think the current implementation keeps things simple. The result of the action is creating a translation from POST values so passing the user_id seems appropriate. Only those with the 'approve' permission can do it so I don't think there's a security concern.